### PR TITLE
Api forwarding drop down

### PR DESCRIPTION
**Pull request by mactanxin**
Fixed Bug #17156

### CODE REVIEW

File: ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js [hunk 1]
```diff
@@ -9,16 +9,28 @@ var Component = require("montage/ui/component").Component,
...
+                this.apiForwardingAddress = [];
             if(isFirstTime) {
                 this.isLoading = true;
                 this.application.dockerSettingsService.getDockerHostQueryData().then(function(dockerHosts) {
                     self.dockerHosts = dockerHosts;
                 })
                 this.application.dockerSettingsService.getDockerConfigData().then(function (dockerConfig) {
-                    self.object = dockerConfig
+                    self.object = dockerConfig;
+                    var api_forwarding_list = dockerConfig.api_forwarding.split(",")
```
Comment (pchaussalet) [inline ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js]: Check if `dockerConfig.api_forwarding` is defined and is it is a string before calling `.split`

File: ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js [hunk 2]
```diff
@@ -9,16 +9,28 @@ var Component = require("montage/ui/component").Component,
...
+                    if(dockerConfig.api_forwarding && dockerConfig.api_forwarding.length > 0) {
+                        for (var i in api_forwarding_list) {
```
Comment (pchaussalet) [inline ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js]: As of today, `for ... in` constructs in Javascript are known to have bad performances, please use a classical `for` loop instead

File: ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js [hunk 3]
```diff
@@ -9,16 +9,31 @@ var Component = require("montage/ui/component").Component,
  * @extends Component
  */
 exports.DockerSettings = Component.specialize(/** @lends DockerSettings# */ {
+
+    apiForwardingAddress: {
```
Comment (pchaussalet) [inline ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js]: This property doesn't seems to be used anymore.

File: ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js [hunk 4]
```diff
@@ -9,16 +9,26 @@ var Component = require("montage/ui/component").Component,
...
+
     enterDocument: {
         value: function(isFirstTime) {
             var self = this;
             if(isFirstTime) {
                 this.isLoading = true;
                 this.application.dockerSettingsService.getDockerHostQueryData().then(function(dockerHosts) {
-                    self.dockerHosts = dockerHosts;
+                    if (dockerHosts.length > 0) {
+                        for (var i = 0; i < dockerHosts.length; i++) {
```
Comment (thibaultzanini) [inline ui/inspectors/system-section.reel/docker-settings.reel/docker-settings.js]: `if` condition not needed here, can be transform into: 
`for (var i = 0, length = dockerHosts.length; i < length; i++) {
...
}`

**Pull request merged**